features/netsocket/TLSSocket.h

```diff
@@ -75,6 +75,9 @@ class TLSSocket : public TLSSocketWrapper {
      *  Initiates a connection to a remote server specified by either
      *  a domain name or an IP address and a port.
      *
+     *  @note: In case connect() returns an error, the state of the socket is
+     *  unspecified. A new socket should be created before reconnecting.
```

Comment:
> Basically, if SSL handshake fails, we return NSAPI_ERROR_AUTH_FAILURE

I noticed we sometimes return PARAMETER_ERROR instead, but I saw no reason to do that and I modified the return value. Please take a look and if you agree with that change, I will amend the test's expectations (the tests documentation just says "must return an error", without specifying the exact error code, so that's fine).
If we always return AUTH_ERROR on ssl-related errors we can safaely tell the users that if they see this error, then they must get a new socket before retrying (which means they don't have to do that if they just want to retry the TCP connection).
Is that ok?